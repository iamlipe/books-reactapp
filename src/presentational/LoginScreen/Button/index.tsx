import React, {memo} from 'react';
import styled, {useTheme} from 'styled-components/native';

interface Props {
  title: string;
  loading?: boolean;
  onPress: () => void;
}

const Button = ({title, loading = false, onPress}: Props) => {
  const theme = useTheme();

  return (
    <StyledContainer onPress={onPress}>
      {loading ? (
        <StyledLoading size="small" color={theme.colors.PRIMARY} />
      ) : (
        <StyledText>{title}</StyledText>
      )}
    </StyledContainer>
  );
};

const StyledContainer = styled.TouchableOpacity`
  width: 85px;
  height: 35px;

  align-items: center;
  justify-content: center;

  border-radius: 44px;

  background-color: ${({theme}) => theme.colors.WHITE};
`;

const StyledText = styled.Text`
  color: ${({theme}) => theme.colors.PRIMARY};
`;

const StyledLoading = styled.ActivityIndicator``;

export default memo(Button);
