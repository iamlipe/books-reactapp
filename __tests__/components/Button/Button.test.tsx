import React from 'react';
import {
  renderWithThemeProvider,
  matchSnapshotWithProvider,
} from '@config/functions/Testing';

import Button from '@presentational/LoginScreen/Button';

describe('Button', () => {
  const title = 'title';
  const onPress = jest.fn();

  test('should render correctly', () => {
    renderWithThemeProvider(<Button title={title} onPress={onPress} />);
  });

  test('should match snapshot', () => {
    matchSnapshotWithProvider(<Button title={title} onPress={onPress} />);
  });

  test('should render title correctly', () => {
    const button = renderWithThemeProvider(
      <Button title={title} onPress={onPress} />,
    );

    const titleButton = button.queryByText(title);

    expect(titleButton).toBeTruthy();
  });

  test('should render loading', () => {
    const button = renderWithThemeProvider(
      <Button title={title} onPress={onPress} loading />,
    );

    const loadingButton = button.queryByTestId('loading-button');

    expect(loadingButton).toBeTruthy();
  });
});
