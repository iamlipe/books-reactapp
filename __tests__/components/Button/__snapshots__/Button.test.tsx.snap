// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderRadius": 44,
      "height": 35,
      "justifyContent": "center",
      "opacity": 1,
      "width": 85,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#B22E6F",
        },
      ]
    }
  >
    title
  </Text>
</View>
`;
